### PR TITLE
fix: setProviderAndWait does not hang on ProviderError

### PR DESCRIPTION
**Pull request by fabriziodemaria**
## This PR
`setContextAndWait` shouldn't hang if the Provider fails the initalization phase for some reason.

### Related Issues
N/A

### Follow-up Tasks
Similar logic will need to be added to setEvaluationContext 

### How to test
Added unit tests

### CODE REVIEW

File: android/src/main/java/dev/openfeature/sdk/OpenFeatureAPI.kt [hunk 1]
```diff
@@ -11,7 +11,11 @@ object OpenFeatureAPI {
     fun setProvider(provider: FeatureProvider, initialContext: EvaluationContext? = null) {
         this@OpenFeatureAPI.provider = provider
         if (initialContext != null) context = initialContext
-        provider.initialize(context)
+        try {
+            provider.initialize(context)
+        } catch (e: Throwable) {
+            // This should never happen
```
Comment (nicklasl) [inline android/src/main/java/dev/openfeature/sdk/OpenFeatureAPI.kt:17]: Can we rephrase this to 
> This is not allowed to happen.

and maybe reference the spec?

Comment (fabriziodemaria) [reply android/src/main/java/dev/openfeature/sdk/OpenFeatureAPI.kt]: I don't think there is anything about this in the Spec. It's this specific implementation that expects the Provider to never throw and rather reads ProviderError events from the Provider.
The Java SDK, for example, works in the opposite way and expects the Provider to throw [in case of errors](https://github.com/open-feature/java-sdk/blob/1b6e16017783aa45ece0c7a6b5ba0f01100813c9/src/main/java/dev/openfeature/sdk/ProviderRepository.java#L140C16-L140C16)

File: android/src/test/java/dev/openfeature/sdk/helpers/SlowProvider.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.launch
+
+class SlowProvider(override val hooks: List<Hook<*>> = listOf(), private var dispatcher: CoroutineDispatcher) : FeatureProvider {
+    override val metadata: ProviderMetadata = SlowProviderMetadata("Slow provider")
+    private var ready = false
+    private var eventHandler = EventHandler(dispatcher)
+    override fun initialize(initialContext: EvaluationContext?) {
+        CoroutineScope(dispatcher).launch {
+            delay(10000)
```
Comment (nicklasl) [inline android/src/test/java/dev/openfeature/sdk/helpers/SlowProvider.kt:25]: nice!

**Review by nicklasl — APPROVED** [review]
